File: Backend/user-service/spring_boot_backend_template/spring_boot_backend_template/src/main/java/com/backend/entity/BusSchedule.java
package com.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonManagedReference;

@Entity
@Table(name = "bus_schedules")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class BusSchedule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String fromCity;

    @Column(nullable = false)
    private String toCity;

    @Column(nullable = false)
    private LocalDate journeyDate;

    @Column(nullable = false)
    private LocalTime departureTime;

    @Column(nullable = false)
    private LocalTime arrivalTime;

    @Column(nullable = false)
    private double ticketPrice;

    @ManyToOne
    @JoinColumn(name = "bus_id", nullable = false)
    private Bus bus;
    
    
 // ONE BUS SCHEDULE can have MANY boarding points
    @OneToMany(mappedBy = "busSchedule", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JsonManagedReference
    private List<BoardingPoint> boardingPoints;

    // ONE BUS SCHEDULE can have MANY dropping points
    @OneToMany(mappedBy = "busSchedule", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JsonManagedReference
    private List<DroppingPoint> droppingPoints;
}

File: Backend/admin-service/admin/admin-service/Models/BusSchedule.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace admin_service.Models
{
    [Table("bus_schedules")]
    public class BusSchedule
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        [Column("from_city")]
        public string FromCity { get; set; }

        [Required]
        [Column("to_city")]
        public string ToCity { get; set; }

        [Required]
        [Column("journey_date")]
        public DateTime JourneyDate { get; set; } // Only date part will be used

        [Required]
        [Column("departure_time")]
        public TimeSpan DepartureTime { get; set; } // Use TimeSpan for time

        [Required]
        [Column("arrival_time")]
        public TimeSpan ArrivalTime { get; set; } // Use TimeSpan for time

        [Required]
        [Column("ticket_price")]
        public double TicketPrice { get; set; }

        // Foreign key relationship
        [Required]
        [ForeignKey("Bus")]
        [Column("bus_id")]
        public long BusId { get; set; }

        public Bus Bus { get; set; }
    }
}
